app/views/organized/exports/show.html.haml

```diff
@@ -45,7 +45,9 @@
 
             .collapse#show_exported_bikes
               %ul.exported-link-list
-                %li= bike_ids.map { |id| link_to(id, bike_path(id)) }.join(", ").html_safe
+                - bike_ids.each_with_index do |id, i|
+                  %li
+                    #{link_to id, bike_path(id)}#{i == (@export.exported_bike_ids.count - 1) ? "" : ","}
```

Comment:
@jmromer I'm not sure why you removed this in #999 ?

First, by switching to a map, it put everything in one `li`, which is a bug. And this `each_with_index` is a simple thing to add commas after the links. It's pretty hacky, but I don't see a reason to drop it?

Other than those actual functional issues, while I'm a fan of `map` in general, my impulse is to avoid using `.html_safe` whenever possible, because I think it's a good habit. I'm open to having this opinion changed though.

Reply:
> First, by switching to a map, it put everything in one li, which is a bug

The regression here is the misplaced `li`, not the introduction of the `map` — these are (strictly speaking) independent of each other, although to the extent the one impacts the other I'd argue that the added complexity of the hackier approach increases the risk of, e.g., misplacing an `li`, by undermining the legibility of the code here. 

(The catalyst for refactoring here in the first place is having to prep for the externalization script, although I think I may have figured out a less intrusive workaround at this point, so I should be able to leave these comma-adding loops alone moving forward.)  

> my impulse is to avoid using .html_safe

I'd assume that "avoid" here is subject to the constraint of not introducing hacky code, which I'd take as an overriding concern. 

That said, I'm close to indifferent between the two approaches if you want to revert to the loop. In my reasoning here I'm also taking as given the requirement of commas and the use of `li`s here.

Reply:
Correct, the missing `li` being the regression, not the map. 

I understood the catalyst. And it’s impressive so much has been updated with so few issues!

“Avoid” is pretty strong in this context, because `html_safe` opens up security issues. If there is any user modifiable content that gets wrapped in html_safe, we have a problem and the cognitive load of trying to think of ways in which someone could manage to modify content is large enough to overweigh fairly ugly stuff.

I think the truly non-hacky solution is done with `map` and something to do with content tags so you don’t need `html_safe`, then joining with commas. But I don’t know exactly how to do that, so I’m going to merge this.

Reply:
[`safe_join`](https://apidock.com/rails/ActionView/Helpers/OutputSafetyHelper/safe_join) is another approach:

```rb
links = [helper.link_to('link-one', ''), helper.link_to('link-two', '')]

links.join(", ".html_safe).html_safe?
# false

joined_str = helper.safe_join(links, ", ")
joined_str.html_safe?
# true
```

Reply:
Oh HELL yeah. 💚